### ISSUE TITLE
Improper handling of exclamation mark in strings?

### ISSUE THREAD

**Issue by halvorlu** [rosenbrockc/fortpy#84]
It seems that the parser mistakes exclamation marks inside a string as starts of comments, which causes the parsing to fail. Consider the following file:

```fortran
module mymodule
contains
  subroutine test()
    call other_sub('hello!')
  end subroutine test
end module mymodule
```

This causes a `(error "\"UnboundLocalError(\\\"local variable 'dependent' referenced before assignment\\\",)\"")` as seen from emacs. I tried delving into the code myself, but it's probably quicker to fix for the code's original author! The error seems to originate from line 291 in fortpy/parsers/executable.py.

**Comment by rosenbrockc** [comment]
It was a simple fix. Available from Revision 1.7.4, which I just posted to `PyPI`.

**Comment by halvorlu** [comment]
Thanks for resolving this!
